src/Apps/Order/Routes/Payment/index.tsx

```diff
@@ -83,79 +111,116 @@ export class PaymentRoute extends Component<PaymentProps, PaymentState> {
+                              onSelect={hideBillingAddress =>
+                                this.setState({ hideBillingAddress })
+                              }
+                            >
+                              Billing and shipping addresses are the same
+                            </Checkbox>
+                          )}
+                          <Collapse
+                            open={
+                              this.isPickup() || !this.state.hideBillingAddress
+                            }
+                          >
+                            <AddressForm
+                              defaultValue={values}
+                              errors={errors}
+                              touched={touched}
+                              onChange={formikHandleChange}
+                              onBlur={handleBlur}
+                              billing
+                            />
+                          </Collapse>
+                          {!xs && (
+                            <ContinueButton
+                              onClick={handleSubmit}
+                              loading={isComittingMutation}
+                            />
+                          )}
+                        </Join>
+                        <Spacer mb={3} />
+                      </>
+                    }
+                    Sidebar={
+                      <Flex flexDirection="column">
+                        <TransactionSummary order={order} mb={[2, 3]} />
+                        <Helper
+                          artworkId={order.lineItems.edges[0].node.artwork.id}
+                        />
+                        {xs && (
+                          <>
+                            <Spacer mb={3} />
+                            <ContinueButton
+                              onClick={handleSubmit}
+                              loading={isComittingMutation}
+                            />
+                          </>
+                        )}
+                      </Flex>
+                    }
                   />
-                  {xs && (
-                    <>
-                      <Spacer mb={3} />
-                      <ContinueButton
-                        onClick={this.onContinueButtonPressed}
-                        loading={isComittingMutation}
-                      />
-                    </>
-                  )}
-                </Flex>
-              }
-            />
+                )}
+              </Responsive>
+            </form>
           )}
-        </Responsive>
+        </Formik>
       </>
     )
   }
 
-  private getSelectedBillingAddress(): stripe.TokenOptions {
+  // Infer the billing address from the form or shipping address (maybe not needed)
+  private getSelectedBillingAddress(formAddress: Address): Address {
+    return (this.isPickup() || !this.state.hideBillingAddress
+      ? formAddress
+      : this.props.order.requestedFulfillment) as Address
+  }
```

Comment:
maybe not needed separate from `getStripeBillingAddress()` below.

Reply:
yeah I think you can just inline it.